### PR TITLE
Jsonify

### PR DESCRIPTION
**Pull request by airconditioner**
Code review

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -58,3 +58,7 @@ Thumbs.db
 *.README*
 *.access
 *.launch
+*.metadata/*
```
Comment (jtnelson) [inline .gitignore]: Can you remove your .gitignore from the commit?

File: concourse-server/src/main/java/org/cinchapi/concourse/server/ConcourseServer.java [hunk 2]
```diff
@@ -1050,6 +1052,59 @@ else if(symbol instanceof Expression) {
...
+     * @param store
+     * @param jsonStr
+     * @return the AtomicOperation
+     */
+    private AtomicOperation doJsonify(List<Long> records, boolean 
+    		includePrimaryKey, Compoundable store, StringBuilder jsonStr) {
+    	AtomicOperation operation = store.startAtomicOperation();
+    	try {
+	   		List<Map<String, Set<Object>>> result = Lists.newLinkedList();
+			Set<Map<String, Set<Object>>> resultNoKey = Sets.newHashSet();
```
Comment (jtnelson) [inline concourse-server/src/main/java/org/cinchapi/concourse/server/ConcourseServer.java]: Too much indentation in this block

File: concourse-server/src/main/java/org/cinchapi/concourse/server/storage/Functions.java [hunk 3]
```diff
@@ -36,6 +37,18 @@
  * @author jnelson
  */
 public final class Functions {
+	
+	
+	/**
+	 * A function that transforms a {@link TObject} to a {@link Java Object}.
+	 */
+	public static final Function<TObject, Object> TOBJECT_TO_JAVA = new Function<TObject, Object>() {
```
Comment (jtnelson) [inline concourse-server/src/main/java/org/cinchapi/concourse/server/storage/Functions.java]: Can you call this function `THRIFT_TO_JAVA` for consistency with the analgous method in the Convert class?

**Comment by jtnelson** [comment]
Your ConcourseServer diff is really large. Can you format the file in Eclipse (CTRL + F)?

File: concourse/src/main/java/org/cinchapi/concourse/util/Convert.java [hunk 4]
```diff
@@ -143,6 +170,9 @@ else if(object instanceof Tag) {
                     "The JSON string must encapsulate data within an object");
         }
         JsonObject object = (JsonObject) parser.parse(json);
+        if (object.has("$primaryKey$")) {
+        	object.remove("$primaryKey$");
```
Comment (jtnelson) [inline concourse/src/main/java/org/cinchapi/concourse/util/Convert.java]: We should add a validation rule that prevents anyone from adding this key directly

File: concourse-server/src/main/java/org/cinchapi/concourse/server/ConcourseServer.java [hunk 5]
```diff
@@ -1050,6 +1052,59 @@ else if(symbol instanceof Expression) {
...
+				for (Map.Entry<String, Set<TObject>> entry: r.entrySet()) {
+					Set<Object> values = Transformers.transformSet(entry.getValue(), 
+							Functions.TOBJECT_TO_JAVA);
+					rec.put(entry.getKey(), values);
+				}
+				if (includePrimaryKey) {
+					Set<Object> recordSet = Sets.newHashSet();
+					recordSet.add(record);
+					rec.put("$primaryKey$", recordSet);
+					result.add(rec);
```
Comment (jtnelson) [inline concourse-server/src/main/java/org/cinchapi/concourse/server/ConcourseServer.java]: Should define this statically somewhere. Adding the value to GlobalState would work

**Comment by airconditioner** [comment]
Sorry I think I was having some indentation issues with my new Eclipse installation. It is causing extra tabs to show up in the saved files. I changed my formatting profile's tab policy to white spaces only. Hope that doesn't cause any issues.

I don't quite know why my ConcourseService changes are so large. My terminal does show Thrift version 0.9.2.

Could you also explain a little bit more about the validation rule and adding values to GlobalState? Am I supposed to write a private validation method and put it in all the places where users can add "$primaryKey$"?

**Pull request closed without merging**